plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java

```diff
@@ -656,22 +660,239 @@ public static void registerJavaLangThrowableIntrinsics(CompilationContext ctxt)
+            GlobalVariable gmdVariable = (GlobalVariable) builder.globalVariable(mdTypes.getAndRegisterGlobalMethodData(builder.getCurrentElement()));
+            Value tablePointer = builder.load(builder.memberOf(gmdVariable, gmdType.getMember("methodInfoTable")), MemoryAtomicityMode.UNORDERED);
+
+            ValueHandle minfoHandle = builder.elementOf(builder.pointerHandle(builder.bitCast(tablePointer, minfoType.getPointer())), arguments.get(0));
+            return builder.load(builder.memberOf(minfoHandle, minfoType.getMember("className")), MemoryAtomicityMode.UNORDERED);
+        };
+
+        intrinsics.registerIntrinsic(Phase.LOWER, mdDesc, "getClassName", intToStringDesc, getClassName);
+
+        StaticIntrinsic getMethodName = (builder, target, arguments) -> {
+            GlobalVariable gmdVariable = (GlobalVariable) builder.globalVariable(mdTypes.getAndRegisterGlobalMethodData(builder.getCurrentElement()));
+            Value tablePointer = builder.load(builder.memberOf(gmdVariable, gmdType.getMember("methodInfoTable")), MemoryAtomicityMode.UNORDERED);
+
+            ValueHandle minfoHandle = builder.elementOf(builder.pointerHandle(builder.bitCast(tablePointer, minfoType.getPointer())), arguments.get(0));
+            return builder.load(builder.memberOf(minfoHandle, minfoType.getMember("methodName")), MemoryAtomicityMode.UNORDERED);
+        };
+
+        intrinsics.registerIntrinsic(Phase.LOWER, mdDesc, "getMethodName", intToStringDesc, getMethodName);
+
+        StaticIntrinsic getMethodDesc = (builder, target, arguments) -> {
+            GlobalVariable gmdVariable = (GlobalVariable) builder.globalVariable(mdTypes.getAndRegisterGlobalMethodData(builder.getCurrentElement()));
+            Value tablePointer = builder.load(builder.memberOf(gmdVariable, gmdType.getMember("methodInfoTable")), MemoryAtomicityMode.UNORDERED);
+
+            ValueHandle minfoHandle = builder.elementOf(builder.pointerHandle(builder.bitCast(tablePointer, minfoType.getPointer())), arguments.get(0));
+            return builder.load(builder.memberOf(minfoHandle, minfoType.getMember("methodDesc")), MemoryAtomicityMode.UNORDERED);
+        };
+
+        intrinsics.registerIntrinsic(Phase.LOWER, mdDesc, "getMethodDesc", intToStringDesc, getMethodDesc);
+
+        String methodDataClass = "org/qbicc/runtime/stackwalk/MethodData";
+        MethodElement getLineNumberElement = ctxt.getRuntimeHelperMethod(methodDataClass, "getLineNumber");
+        MethodElement getMethodInfoIndexElement = ctxt.getRuntimeHelperMethod(methodDataClass, "getMethodInfoIndex");
+        MethodElement getFileNameElement = ctxt.getRuntimeHelperMethod(methodDataClass, "getFileName");
+        MethodElement getClassNameElement = ctxt.getRuntimeHelperMethod(methodDataClass, "getClassName");
+        MethodElement getMethodNameElement = ctxt.getRuntimeHelperMethod(methodDataClass, "getMethodName");
+
+        StaticIntrinsic fillStackTraceElement = (builder, target, arguments) -> {
```

Comment:
Could some of these intrinsics be written in Java as "native" methods?  Might be more maintainable.

Reply:
yes, I agree and that was my intention initially and hence created https://github.com/qbicc/qbicc/pull/611 as accessing the native arrays does not work properly. But it has been closed in favor of https://github.com/qbicc/qbicc/issues/695. So once that gets resolved, I can replace these intrinsics with the Java methods.